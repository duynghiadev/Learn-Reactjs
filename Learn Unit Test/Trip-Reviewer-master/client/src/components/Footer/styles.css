.footer-wrapper {
  padding: 15px;
  background-color: #f2f2f2;
  box-shadow: 0px -5px 0px #bfbfbf;
  color: black;
  text-align: center;
}

.footer-wrapper i {
  color: #808080;
  margin-left: 5px;
  margin-right: 5px;
}

.footer-wrapper p {
  margin: 5px;
  font-size: 12px;
}

.footer-wrapper .author {
  font-size: 16px;
  margin-top: 15px;
}

.footer-wrapper a {
  color: white;
  font-size: 12px;
}

.footer-wrapper .social-media-icons i {
  font-size: 25px;
  color: #808080;
  transition: 0.5s ease-in-out;
}

.footer-wrapper .social-media-icons a {
  margin-right: 20px;
}

.footer-wrapper .footer-github:hover i {
  color: #000000;
  text-shadow: 3px 3px 5px gray;
}

.footer-wrapper .footer-facebook:hover i {
  color: #3b5998;
  text-shadow: 3px 3px 5px gray;
}

.footer-wrapper .footer-vk:hover i {
  color: #507299;
  text-shadow: 3px 3px 5px gray;
}

.footer-wrapper .footer-linkedin:hover i {
  color: #0077b5;
  text-shadow: 3px 3px 5px gray;
}
