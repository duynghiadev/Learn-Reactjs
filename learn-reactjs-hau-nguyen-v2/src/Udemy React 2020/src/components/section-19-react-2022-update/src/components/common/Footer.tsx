export interface FooterProps {}

export function Footer(props: FooterProps) {
  return (
    <div>
      <h2>Footer</h2>
    </div>
  )
}
