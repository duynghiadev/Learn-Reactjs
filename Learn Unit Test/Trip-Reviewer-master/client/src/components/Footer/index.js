import React from 'react';

import './styles.css';

const Footer = () => {
  const renderIcon = (url, type) => (
    <a target="_blank" rel="noopener noreferrer" href={url} className={`footer-${type}`}>
      <i className={`fab fa-${type}`} />
    </a>
  );

  return (
    <footer className="footer-wrapper">
      <div className="footer-content">
        <div className="social-media-icons">
          {renderIcon('https://github.com/KovDimaY/Trip-Reviewer', 'github')}
          {renderIcon('https://www.facebook.com/dmytro.kovalenko.1004', 'facebook')}
          {renderIcon('https://www.vk.com/id11545172', 'vk')}
          {renderIcon('https://www.linkedin.com/in/kovalenkodmytro', 'linkedin')}
        </div>
        <p className="author">Copyright &copy; Dmytro Kovalenko ;)</p>
      </div>
    </footer>
  );
};

export default Footer;
