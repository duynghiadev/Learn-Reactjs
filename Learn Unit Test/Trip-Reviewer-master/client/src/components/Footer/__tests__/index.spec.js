import React from 'react';
import { create } from 'react-test-renderer';

import Footer from '..';

const mockComponent = props => <Footer {...props} />;

describe('<Footer />', () => {
  it('should render component', () => {
    const tree = create(mockComponent()).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
