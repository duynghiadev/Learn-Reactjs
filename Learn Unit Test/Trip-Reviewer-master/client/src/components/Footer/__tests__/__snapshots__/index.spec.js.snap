// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render component 1`] = `
<footer
  className="footer-wrapper"
>
  <div
    className="footer-content"
  >
    <div
      className="social-media-icons"
    >
      <a
        className="footer-github"
        href="https://github.com/KovDimaY/Trip-Reviewer"
        rel="noopener noreferrer"
        target="_blank"
      >
        <i
          className="fab fa-github"
        />
      </a>
      <a
        className="footer-facebook"
        href="https://www.facebook.com/dmytro.kovalenko.1004"
        rel="noopener noreferrer"
        target="_blank"
      >
        <i
          className="fab fa-facebook"
        />
      </a>
      <a
        className="footer-vk"
        href="https://www.vk.com/id11545172"
        rel="noopener noreferrer"
        target="_blank"
      >
        <i
          className="fab fa-vk"
        />
      </a>
      <a
        className="footer-linkedin"
        href="https://www.linkedin.com/in/kovalenkodmytro"
        rel="noopener noreferrer"
        target="_blank"
      >
        <i
          className="fab fa-linkedin"
        />
      </a>
    </div>
    <p
      className="author"
    >
      Copyright Â© Dmytro Kovalenko ;)
    </p>
  </div>
</footer>
`;
